libobs/media-io/audio-resampler-ffmpeg.c

```diff
@@ -99,28 +103,47 @@ audio_resampler_t *audio_resampler_create(const struct resample_info *dst,
 
 	rs->opened = false;
 	rs->input_freq = src->samples_per_sec;
-	rs->input_layout = convert_speaker_layout(src->speakers);
 	rs->input_format = convert_audio_format(src->format);
 	rs->output_size = 0;
 	rs->output_ch = get_audio_channels(dst->speakers);
 	rs->output_freq = dst->samples_per_sec;
-	rs->output_layout = convert_speaker_layout(dst->speakers);
 	rs->output_format = convert_audio_format(dst->format);
 	rs->output_planes = is_audio_planar(dst->format) ? rs->output_ch : 1;
-
+#if (LIBSWRESAMPLE_VERSION_INT < AV_VERSION_INT(4, 5, 100))
+	rs->input_layout = convert_speaker_layout(src->speakers);
+	rs->output_layout = convert_speaker_layout(dst->speakers);
 	rs->context = swr_alloc_set_opts(NULL, rs->output_layout,
 					 rs->output_format,
 					 dst->samples_per_sec, rs->input_layout,
 					 rs->input_format, src->samples_per_sec,
 					 0, NULL);
-
+#else
+	int nb_ch = get_audio_channels(src->speakers);
+	av_channel_layout_default(&rs->input_ch_layout, nb_ch);
+	av_channel_layout_default(&rs->output_ch_layout, rs->output_ch);
+	if (src->speakers == SPEAKERS_4POINT1)
+		rs->input_ch_layout =
+			(AVChannelLayout)AV_CHANNEL_LAYOUT_4POINT1;
+	if (dst->speakers == SPEAKERS_4POINT1)
+		rs->output_ch_layout =
+			(AVChannelLayout)AV_CHANNEL_LAYOUT_4POINT1;
+	swr_alloc_set_opts2(&rs->context, &rs->output_ch_layout,
```

Comment:
Please place a blank line after the contents of unenclosed `if` statements. It makes it clearer that the `if`-specific line has concluded. Otherwise, enclose the contents in braces.

Reply:
done